modules/javafx.graphics/src/main/native-glass/mac/a11y/AccessibleBase.m

```diff
@@ -219,15 +186,13 @@ - (BOOL)performAccessibleAction:(NSString *)action
 
 - (NSRect)accessibilityFrame
 {
-    jobject jresult = NULL;
-    GET_MAIN_JENV;
-    if (env == NULL) return NSZeroRect;
-    jresult = (jobject)(*env)->CallLongMethod(env, self->jAccessible, jAccessibilityAttributeValue, (jlong)@"AXPosition");
-    GLASS_CHECK_EXCEPTION(env);
-    NSPoint position = [variantToID(env, jresult) pointValue];
-    jresult = (jobject)(*env)->CallLongMethod(env, self->jAccessible, jAccessibilityAttributeValue, (jlong)@"AXSize");
-    GLASS_CHECK_EXCEPTION(env);
-    NSSize size = [variantToID(env, jresult) sizeValue];
+    id p = [self requestNodeAttribute:@"AXPosition"];
+    id s = [self requestNodeAttribute:@"AXSize"];
+    if (p == NULL || s == NULL) {
```

Comment:
not a big issue, but would it be (marginally) better to bail out early if p == NULL ?

Reply:
But that would require two conditions - one for position and one for size - with the same logic which would look not that compact and pretty.

Reply:
but it will save a couple of nanoseconds!

the code is ok.  one question though: what are the circumstances when these calls would return NULL?

Reply:
right, thank you for clarification!